### PR TITLE
gofail: show error messages returned by gofail HTTP endpoints

### PR DESCRIPTION
**Pull request by henrybear327**
This PR introduces showing the error message returned by gofail HTTP endpoints, making debugging easier.

Also apply simplified `Enabled()` logic, as suggested by the linter

Please read https://github.com/etcd-io/etcd/blob/main/CONTRIBUTING.md#contribution-flow.

### CODE REVIEW

File: tests/framework/e2e/etcd_process.go [hunk 1]
```diff
@@ -375,7 +375,11 @@ func (f *BinaryFailpoints) SetupHTTP(ctx context.Context, failpoint, payload str
 	}
 	defer resp.Body.Close()
 	if resp.StatusCode != http.StatusNoContent {
-		return fmt.Errorf("bad status code: %d", resp.StatusCode)
+		errMsg, err := io.ReadAll(resp.Body)
+		if err != nil {
+			return err
```
Comment (siyuanfoundation) [inline tests/framework/e2e/etcd_process.go:380]: return statusCode here as well? `fmt.Errorf("bad status code: %d (%v)", resp.StatusCode, err)`

File: tests/framework/e2e/etcd_process.go [hunk 2]
```diff
@@ -404,17 +408,18 @@ func (f *BinaryFailpoints) DeactivateHTTP(ctx context.Context, failpoint string)
...
```
Comment (siyuanfoundation) [inline tests/framework/e2e/etcd_process.go:413]: ditto

File: tests/framework/e2e/etcd_process.go (hunk 1 above)
Comment (henrybear327) [reply tests/framework/e2e/etcd_process.go]: Thanks, fixed. :)

File: tests/framework/e2e/etcd_process.go (hunk 2 above)
Comment (henrybear327) [reply tests/framework/e2e/etcd_process.go]: Fixed

File: tests/framework/e2e/etcd_process.go [hunk 3]
```diff
@@ -404,17 +408,18 @@ func (f *BinaryFailpoints) DeactivateHTTP(ctx context.Context, failpoint string)
...
 	return nil
 }
 
 func (f *BinaryFailpoints) Enabled() bool {
 	_, err := failpoints(f.member)
-	if err != nil {
-		return false
-	}
-	return true
+	return err == nil
```
Comment (ahrtr) [inline tests/framework/e2e/etcd_process.go:422]: Nice cleanup.

File: tests/framework/e2e/etcd_process.go [hunk 4]
```diff
@@ -375,7 +375,11 @@ func (f *BinaryFailpoints) SetupHTTP(ctx context.Context, failpoint, payload str
 	}
 	defer resp.Body.Close()
 	if resp.StatusCode != http.StatusNoContent {
-		return fmt.Errorf("bad status code: %d", resp.StatusCode)
+		errMsg, err := io.ReadAll(resp.Body)
```
Comment (ahrtr) [inline tests/framework/e2e/etcd_process.go:378]: Currently `gofail PUT` doesn't return any data in the body. So can we keep it unchanged for simplicity?

Comment (henrybear327) [reply tests/framework/e2e/etcd_process.go:378]: Sure! Rolling back!

Thanks for reviewing!

Comment (ahrtr) [reply tests/framework/e2e/etcd_process.go]: Sorry, my previous comment isn't correct. `gofail PUT` will write some error message when failed (see link below). Not a big deal. But please keep keep it consistent between `SetupHTTP` and `DeactivateHTTP`. 

https://github.com/etcd-io/gofail/blob/93c579a86c468537617dedd1dc0f63c8f27926ca/runtime/http.go#L69-L81

Comment (henrybear327) [reply tests/framework/e2e/etcd_process.go:378]: Sorry @ahrtr  I pushed the wrong commit at the airport :( 

I checked the gofail source code and saw that both PUT and DELETE will return error messages! Thus the commit will stay as-is.

File: tests/framework/e2e/etcd_process.go [hunk 5]
```diff
@@ -375,7 +375,11 @@ func (f *BinaryFailpoints) SetupHTTP(ctx context.Context, failpoint, payload str
...
+		if err != nil {
+			return fmt.Errorf("bad status code: %d (%v)", resp.StatusCode, err)
+		}
+		return fmt.Errorf("bad status code: %d (%v)", resp.StatusCode, errMsg)
```
Comment (fuweid) [inline tests/framework/e2e/etcd_process.go:382]: I think we should use `string(errMsg)` here becasue `errMsg` is `[]byte`. It's not human-friendly.

Comment (ahrtr) [reply tests/framework/e2e/etcd_process.go:382]: Or just use `%s`, see https://github.com/etcd-io/etcd/pull/17878

Comment (henrybear327) [reply tests/framework/e2e/etcd_process.go]: Fixed accordingly!

Also fixed the err's formatting to using %w

**Review by ahrtr — APPROVED** [review]

**Review by serathius — APPROVED** [review]

**Pull request merged**